### PR TITLE
feat(poe-v2): poe v2 support for funnels

### PR DESCRIPTION
**Pull request by yakkomajuri**
Currently with only one test as I'm focusing on getting all insights done before expanding the test suite. I'll be off a week after the offsite so trying to get as far as possible so we can try this out live with our team.

### CODE REVIEW

File: posthog/queries/funnels/funnel_event_query.py [hunk 1]
```diff
@@ -49,7 +44,7 @@ def get_query(
```
Comment (tiina303) [inline posthog/queries/funnels/funnel_event_query.py:44]: can't comment on line 38: shouldn't we be using `self._person_id_alias` there too - we seem to be using person_id from events table, which is only PoEv1, but not PoEv2 compatible, no?

File: posthog/queries/funnels/funnel_event_query.py [hunk 2]
```diff
@@ -115,17 +110,20 @@ def get_query(
         return query, self.params
 
     def _determine_should_join_distinct_ids(self) -> None:
-        if (
+        non_person_id_aggregation = (
```
Comment (tiina303) [inline posthog/queries/funnels/funnel_event_query.py:113]: same as the other PR - should this be in every file separately?

**Review by tiina303 — APPROVED** [review]

File: posthog/queries/funnels/funnel_event_query.py (hunk 1 above)
Comment (yakkomajuri) [reply posthog/queries/funnels/funnel_event_query.py:44]: yup, good spot!

File: posthog/queries/funnels/funnel_event_query.py (hunk 2 above)
Comment (yakkomajuri) [reply posthog/queries/funnels/funnel_event_query.py:113]: replied there but this is unrelated - I'm currently building this in with how the current system works, can think of separate refactors later.

note though that it's not _always_ the same, and having a way to account for that via an abstraction might end up being more complicated than a bit of repetition here and there

**Pull request merged**